system/lz4_sm/CMakeLists.txt

```diff
@@ -39,6 +39,9 @@ SET_SOURCE_FILES_PROPERTIES( ${SRCS} PROPERTIES LANGUAGE C )
 
 if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG)
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -std=c99")
+  if (CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
Does this need to duplicate the check elsewhere:

  if (((CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8.0.0) OR CMAKE_COMPILER_IS_CLANGXX) AND CMAKE_BUILD_TYPE STREQUAL "Debug")

Reply:
I'm not sure it will cause any problem if it doesn't, unless an old compiler objects to the unrecognized option. But I can fix to be on the safe side

Reply:
Put in a cleaner fix